lib/THC/THCBlas.cu

```diff
@@ -211,17 +202,75 @@ void THCudaBlas_gemm(THCState *state, char transa, char transb, long m, long n,
     THCublasCheck(cublasSgemm(THCState_getCurrentBlasHandle(state), opa, opb, i_m, i_n, i_k, &alpha, a, i_lda, b, i_ldb, &beta, c, i_ldc));
     return;
   }
-  THError("Cublas_gemm only supports m, n, k, lda, ldb, ldc"
+  THError("Cublas_Sgemm only supports m, n, k, lda, ldb, ldc"
           "with the bound [val] <= %d", INT_MAX);
 }
 
-void THCudaBlas_gemmBatched(THCState *state, char transa, char transb, long m, long n, long k,
-                            float alpha, const float *a[], long lda, const float *b[], long ldb,
-                            float beta, float *c[], long ldc, long batchCount)
+#ifdef CUDA_HALF_TENSOR
+void THCudaBlas_Hgemm(THCState *state, char transa, char transb, long m, long n, long k, half alpha, half *a, long lda, half *b, long ldb, half beta, half *c, long ldc)
+{
+  adjustLd(transa, transb, m, n, k, &lda, &ldb, &ldc);
+  cublasOperation_t opa = convertTransToCublasOperation(transa);
+  cublasOperation_t opb = convertTransToCublasOperation(transb);
+
+  if( (m <= INT_MAX) && (n <= INT_MAX) && (k <= INT_MAX) && (lda <= INT_MAX)  && (ldb <= INT_MAX) && (ldc <= INT_MAX) )
+  {
+    int i_m = (int)m;
+    int i_n = (int)n;
+    int i_k = (int)k;
+    int i_lda = (int)lda;
+    int i_ldb = (int)ldb;
+    int i_ldc = (int)ldc;
+
+    // Check for native Hgemm support
+    if (THC_nativeHalfInstructions(state)) {
+      THCublasCheck(cublasHgemm(THCState_getCurrentBlasHandle(state), opa, opb, i_m, i_n, i_k, &alpha, a, i_lda, b, i_ldb, &beta, c, i_ldc));
+    } else {
+      // Simulated Hgemm
+      float fAlpha = THC_half2float(alpha);
+      float fBeta = THC_half2float(beta);
+
+      THCublasCheck(cublasSgemmEx(THCState_getCurrentBlasHandle(state), opa, opb, i_m, i_n, i_k, &fAlpha,
+                                  a, CUBLAS_DATA_HALF, i_lda, b, CUBLAS_DATA_HALF, i_ldb, &fBeta, c, CUBLAS_DATA_HALF, i_ldc));
```

Comment:
This should be changed for CUDA 8.0. I fixed it with #ifdef here : https://github.com/borisfom/cutorch/commit/ccff07f0616e1b5b5767eeb8b01e659f530c0ac9